pkg/controller/certificates/trigger/trigger_controller.go

```diff
@@ -244,3 +248,22 @@ func init() {
 			Complete()
 	})
 }
+
+// A simpler version of RequestMatchesSpec that only returns a boolean.
+//
+// If a nil request is given, meaning that the certificate request does not
+// exist yet, the function will return true. Similarly, if anything goes
+// wrong with decoding the CSR, it will be considered as a mismatch.
+func requestMatchesSpec(req *cmapi.CertificateRequest, spec cmapi.CertificateSpec) bool {
+	if req == nil {
+		// We assume that a nil value means that the certificate request
+		// does not exist yet; in that case, we condider that the
+		// certificate matches the request.
+		return true
+	}
+	mismatches, err := certificates.RequestMatchesSpec(req, spec)
+	if err != nil {
+		return false
+	}
+	return len(mismatches) == 0
+}
```

Comment:
Yesterday, @wallrj made a good point about the two early `return` statements; they a bit feel arbitrary:

1. if the certificate request is nil, why would we assume that the certificate "matches" the request when the request doesn't even exist?
2. if the `RequestMatchesSpec` function returns an error, why would we assume that the certificate does not match the certificate request?

I think I'll remove this helper function since it kind of offuscate things... what do you think?

Reply:
Done! I removed this confusing helper function and moved the logic "inline" with where it is being used.